docs/sequence/PISPLinkingApi.puml

```diff
@@ -0,0 +1,577 @@
+  callbackURI: 'pisp-app://callback...',
+  authURI: 'dfspa.com/authorize?consentRequestId=6789' // this is new
+}
+end note
+Switch --> DFSP_LM: 202 ACCEPTED
+deactivate DFSP_LM
+activate PISP_TP_IN
+Switch ->  PISP_TP_IN: PUT /consentRequests/6789
+PISP_TP_IN --> Switch: 200 OK
+deactivate Switch
+activate PISP_LM
+PISP_TP_IN --> PISP_LM: Consent Request response recieved
+rnote right of PISP_LM: state: OTPAuthenticationChannelResponseRecieved
+PISP_LM --> PISP_LM: this.saveToKVS()
+deactivate PISP_TP_IN
+PISP_LM -> PISP_TP_OUT: return Authentication Response
+deactivate PISP_LM
+PISP_TP_OUT --> PISP: 200 OK Autentication Response
+rnote left of PISP_TP_OUT #LightBlue
+{
+  channelResponse: {
+    consentRequestId: 6789
+    scopes: [{
+      accountId: 'dfspa.username.1234',
+      actions: ['accounts.getBalance', 'accounts.transfer'],
+      accountId: 'dfspa.username.5678',
+      actions: ['accounts.getBalance', 'accounts.transfer'],
+    }],
+    authChannels: ["OTP"],
+    callbackURI: 'pisp-app://callback...',
+    authURI: 'dfspa.com/authorize?consentRequestId=6789' // this is new
+  },
+  currentState="OTPAuthenticationChannelResponseRecieved"
+}
+end note
+
+deactivate PISP_TP_OUT
+deactivate PISP
+
+note over PISP, DFSPAuthorizeSimulator
+  The DFSP is expected to send an OTP to the end user which will be used in the next linking step.
+end note
+
+== Authentication+Grant Consent+Register Credential ==
+autonumber 1 "<b>AUTHENTICATION-#</b>"
+rnote right of PISP
+PISP has obtained authToken from end-user(OTP) or through a callback(Web).
+end note
+PISP -> PISP_TP_OUT: POST /linking/authenticate
+rnote right of PISP #LightBlue
+PATCH /linking/authenticate/6789
+{
+  authToken: '123456'
+}
+end note
+
+activate PISP
+activate PISP_TP_OUT
+
+PISP_TP_OUT -> PISP_TP_OUT: const model = await loadFromKVS({key: 6789})
+rnote right of PISP_TP_OUT: state: webAuthenticationChannelResponseRecieved or OTPAuthenticationChannelResponseRecieved
+PISP_TP_OUT -> PISP_LM: model.authenticate()
+
+activate PISP_LM
+
+PISP_LM -> PISP_LM: ThirdpartyRequests.patchConsentRequests()
+PISP_LM -> Switch: PATCH /consentRequests/6789
+rnote right of PISP_LM #LightBlue
+PATCH /consentRequests/6789
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+{
+  authToken: '124356'
+}
+end note
+
+activate Switch
+Switch --> PISP_LM: 202 Accepted
+deactivate PISP_LM
+Switch -> DFSP_TP_IN: PATCH /consentRequests/6789
+activate DFSP_TP_IN
+
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS({key: 6789})
+activate DFSP_LM
+rnote right of DFSP_TP_IN: state: consentRequestValidatedAndStored
+DFSP_TP_IN -> DFSP_LM: model.validateAuthToken()
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+deactivate DFSP_TP_IN
+
+DFSP_LM -> DFSP_LM: DFSPBackendRequests.validateAuthToken()
+rnote right of DFSP_LM #LightBlue
+Do we need two backend endpoints for validating
+web authTokens and OTP authTokens? Or is a DFSP expected to
+validate both cases with one endpoint?
```

Comment:
let propagate to the call payload the `authChannel` value which should be stored in the previous step when callback `PUT /consentReqeusts/{consentRequestId}` was called.

Reply:
when we send `authChannel` then we can easily mock simulator for proper behaviours and testing various scenarios